app/uk/gov/hmrc/agentservicesaccount/services/HostnameWhiteListService.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentservicesaccount.services
+
+import java.net.URL
+import javax.inject.{Inject, Singleton}
+
+import uk.gov.hmrc.agentservicesaccount.AppConfig
+import uk.gov.hmrc.agentservicesaccount.connectors.SsoConnector
+import uk.gov.hmrc.play.binders.ContinueUrl
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+import scala.util.Try
+
+@Singleton
+class HostnameWhiteListService @Inject()(appConfig: AppConfig, ssoConnector: SsoConnector) {
```

Comment:
I find this advice practical: "A **reasonable combination** of unit and integration tests ensures that every single unit works correctly, independently from others, and that all these units play nicely when integrated, giving us a high level of confidence that the whole system works as expected. " from [how-to-write-testable-code](https://www.toptal.com/qa/how-to-write-testable-code-and-why-it-matters)